### PR TITLE
Update for Swift 5

### PR DESCRIPTION
**Pull request by saagarjha**
- [X] This change has been discussed with the author.
- [ ] It implements / fixes issue #.

---

Since Xcode 10.2 dropped yesterday, I think we should merge this in as soon as we get this looked over again. I think there are some `UnsafePointer` APIs left that I should look at; other than that it's mostly boilerplate.

### CODE REVIEW

**Comment by saagarjha** [comment]
(We will also need a new libmpv compiled without Swift, FWIW)

**Comment by Noctem** [comment]
Seems to be working smoothly with the same build tweaks that I normally make (0fe559ef119dfc128f4aeab4270f6d8ceb624d55) and these formulae for [ffmpeg](https://github.com/Noctem/homebrew-custom/blob/master/ffmpeg-iina.rb) and [mpv](https://github.com/Noctem/homebrew-custom/blob/master/mpv-iina.rb).

I still intermittently experience hangs/crashes when a video is launched directly to fullscreen (probably #2221), but I've been experiencing that for months. The conditions for triggering it don't seem to be entirely consistent however.

File: Podfile [hunk 1]
```diff
@@ -5,10 +5,10 @@ target 'iina' do
   use_frameworks!
 
   # Pods for iina
-  pod 'Just', :git => 'https://github.com/JustHTTP/Just.git'
-  pod 'PromiseKit'
+  pod 'Just', :git => 'https://github.com/iina/Just', :branch => 'swift-5'
+  pod 'PromiseKit', :git => 'https://github.com/iina/PromiseKit', :branch => 'xcode-10.2-beta1'
```
Comment (Noctem) [inline Podfile]: Upstream added Swift 5 support and seems to work for me.

Comment (saagarjha) [reply Podfile]: Thanks for letting me know they've added this! I'll update this to point back at the upstream repository.

**Comment by saagarjha** [comment]
Figured out how to build the libraries myself and double-checked that IINA at least runs. Moving out of draft status and requesting review.

**Pull request merged**